Peoplesoft/modules/network/subnets/subnets.tf

```diff
@@ -8,17 +8,41 @@ resource "azurerm_subnet" "subnet" {
   virtual_network_name = "${var.vnet_name}"
   address_prefix       = "${element(values(var.subnet_cidr_map),count.index)}"
   count = "${length(var.subnet_cidr_map)}"
+  network_security_group_id = "${element(values(var.nsg_ids),count.index)}"
+}
+
+locals {
+    # a temp map to avoid a dependency between the order in which subnets show up in
+    # azurerm_subnet.subnet.*.id and the keys in subnet_cidr_map/nsg_ids.
+    subnetNameToID = "${zipmap(keys(var.subnet_cidr_map),azurerm_subnet.subnet.*.id)}"
```

Comment:
What is this for again? It's unclear...

Reply:
Its to help make sure the subnets get matched up with the right NSGs.  It was a shawn thing that existed in Retail but was missing in Peoplesoft and I wasn't sure if it was related to subnet to NSG issue I was having earlier.

Reply:
i'm not sure we need this, tbh. I think ordering is taken care of. But, it doesn't look like it's doing any harm, so, we can keep it for now